src/components/pages/account/TransactionsTable.tsx

```diff
@@ -86,98 +107,75 @@ const columns: ColumnDef<Split>[] = [
   {
     header: 'Actions',
     enableSorting: false,
-    cell: ({ row }) => {
-      const tx = row.original.transaction;
-      const originalSplit = tx.splits.find(split => split.guid === row.original.guid) as Split;
-      const defaultValues = {
-        ...tx,
-        date: tx.date.toISODate() as string,
-        fk_currency: tx.fk_currency as Commodity,
-        // This is hacky but if we pass the Split
-        // class to the form, then we have reference errors as when
-        // we update the form, it also updates the defaultValues
-        // which means formState.isDirty is not triggered properly
-        splits: [
-          {
-            ...originalSplit,
-            value: originalSplit.value,
-            quantity: originalSplit.quantity,
-          },
-          ...tx.splits.filter(split => split.guid !== originalSplit.guid).map(split => ({
-            ...split,
-            value: split.value,
-            quantity: split.quantity,
-          } as Split)),
-        ] as Split[],
-      };
-      return (
-        <>
-          <FormButton
-            id="edit-tx"
-            modalTitle="Edit transaction"
-            buttonContent={<BiEdit className="flex" />}
-            className="link"
-          >
-            <TransactionForm
-              action="update"
-              defaultValues={defaultValues}
-            />
-          </FormButton>
-          <FormButton
-            id="delete-tx"
-            modalTitle="Confirm you want to remove this transaction"
-            buttonContent={<BiXCircle className="flex" />}
-            className="link"
-          >
-            <TransactionForm
-              action="delete"
-              defaultValues={defaultValues}
-            />
-          </FormButton>
-        </>
-      );
-    },
+    cell: ActionsCell,
   },
 ];
 
-function FromToAccountPartial(
-  accounts: AccountsMap,
-) {
-  return function FromToAccount({ row }: CellContext<Split, unknown>): JSX.Element {
-    const { splits } = row.original.transaction;
-    const otherSplits = splits.filter(split => split.account.guid !== row.original.account.guid);
+function useTransaction(guid: string): UseQueryResult<Transaction> {
```

Comment:
So this is quite cool. Instead of having to pre-load a bunch of data so I have all transactions available when I need them, we load the data dynamically using this hook for the Cells that need it.